tree/ntuple/v7/inc/ROOT/REntry.hxx

```diff
@@ -79,6 +79,8 @@ private:
    std::vector<RFieldBase::RValue> fValues;
    /// For fast lookup of token IDs given a (sub)field name present in the entry
    std::unordered_map<std::string, std::size_t> fFieldName2Token;
+   /// To ensure that the entry is standalone, a copy of all field types
+   std::vector<std::string> fFieldTypes;
```

Comment:
This caches allows but does not ensure (in particular future code) to not be reaching out to the model.   Notably the code on (new) line 147 still uses `fValues[..].GetField()` (which uses the model indirectly, isn't it).  Albeit it is 'only' in the error case.  Am I missing something?

Reply:
Yes indeed. In the future this will be dealt with by nulling the `RFieldBase` pointers when destructing the model. For the noted case, I added a function look up the field name in `fFieldName2Token` - since it's only in the error case, linear complexity is fine.